### PR TITLE
Improve hooks table visualisation

### PR DESCRIPTION
**Pull request by olyism**
Added design improvements to better communicate job lifecycle hooks.

The [current table](http://buildkite.com/docs/agent/v3/hooks#job-lifecycle-hooks) doesn't provide an obvious visual representation of the order of hook execution.

🎦 Here's a Loom recording of me going through the changes https://www.loom.com/share/92f99e8ea1e7419c86ac835a6c92773a

Visual design improvements:
- Placed Location Orders in a vertical manner to visualise order
- Append colored icons to help readers to visually differentiate between the different locations
- Split Plugin into non-vendored and vendored for better clarity

Other improvements:
- Improvement maintainability: refactored existing table from html to markdown
- Re-skinned the [agent lifecycle hooks table](http://buildkite.com/docs/agent/v3/hooks#job-lifecycle-hooks) with the same style

# Before

Screen Shot 2022-05-11 at 2 34 00 pm

# After

Screen Shot 2022-05-11 at 2 29 20 pm

### CODE REVIEW

**Comment by olyism** [comment]
Oops. Looks like I have some linting errors to fix. Will work on that.

**Comment by plaindocs** [comment]
I took the liberty of jumping in to your PR to kick the linter.

**Review by plaindocs — APPROVED** [review]
Love this work, huge improvement.

File: pages/agent/v3/hooks.md.erb [hunk 1]
```diff
@@ -131,7 +135,9 @@ hooks defined here will be run for every pipeline that uses the repository.
 * Plugin hooks - these are provided by any [plugins](/docs/plugins) you’ve
 included in your pipeline steps and can define job lifecycle hooks. Job
 lifecycle hooks defined by a plugin will only be run for the step that includes
-them.
+them. Plugins can be *vendored* (if they are already present in the repository,
+and included using a relative path) or *non-vendored* (when they are included
+from GitHub), which affects the order they are run in.
```
Comment (pzeballos) [inline pages/agent/v3/hooks.md.erb:140]: It's not exclusive of Github, non-vendored can be hosted on others repository providers

**Review by pzeballos — CHANGES_REQUESTED** [review]
Just a bit clarification, but the rest looks good!

File: pages/agent/v3/hooks.md.erb [hunk 2]
```diff
@@ -194,7 +135,9 @@ hooks defined here will be run for every pipeline that uses the repository.
...
```
Comment (plaindocs) [inline pages/agent/v3/hooks.md.erb:140]:
```suggestion
from elsewhere), which affects the order they are run in.
```

File: pages/agent/v3/hooks.md.erb [hunk 3]
```diff
@@ -194,7 +135,9 @@ hooks defined here will be run for every pipeline that uses the repository.
 * Plugin hooks - these are provided by any [plugins](/docs/plugins) you’ve
 included in your pipeline steps and can define job lifecycle hooks. Job
 lifecycle hooks defined by a plugin will only be run for the step that includes
-them.
+them. Plugins can be *vendored* (if they are already present in the repository,
```
[the remark this answers is not in the document]
Comment (karensawrey) [reply pages/agent/v3/hooks.md.erb:138]: Clarified. All clear.

File: styleguide/STYLE.md [hunk 4]
```diff
@@ -349,6 +349,13 @@ Line 3, column 1  | Line 3, column 2
 
 The `{: class="two-column"}` class added at the end of a two-column table is what allows the custom table style to work.
 
+#### Prepending icons
+You can prepend an icon to boost the visual emphasis for an inline text. To do this, simply wrap the text with `<span class="add-icon-#{ICON_NAME}">`.
```
Comment (karensawrey) [inline styleguide/STYLE.md:353]:
```suggestion
You can prepend an icon to boost the visual emphasis for an inline text. To do this, wrap the text with `<span class="add-icon-#{ICON_NAME}">`. 
```

File: styleguide/STYLE.md [hunk 5]
```diff
@@ -349,6 +349,13 @@ Line 3, column 1  | Line 3, column 2
 
 The `{: class="two-column"}` class added at the end of a two-column table is what allows the custom table style to work.
 
+#### Prepending icons
+You can prepend an icon to boost the visual emphasis for an inline text. To do this, simply wrap the text with `<span class="add-icon-#{ICON_NAME}">`. 
+
+At the time of writing there are only three icons available — agent, repository and plugin. To add more icons see `$icons` in `_add-icon.scss`, add a new name as the key and the inline SVG. Icon dimension must be 22px * 22px.
```
Comment (karensawrey) [inline styleguide/STYLE.md:355]:
```suggestion
At the time of writing there are only three icons available — agent, repository, and plugin. To add more icons see `$icons` in `_add-icon.scss`, add a new name as the key and the inline SVG. Icon dimension must be 22px * 22px.
```

**Review by karensawrey — CHANGES_REQUESTED** [review]

**Comment by karensawrey** [comment]
> @olyism I reckon we can address Karen's concerns and get this merged in, what do you reckon?

Let's correct & :shipit:

**Comment by olyism** [comment]
:shipit:

**Pull request merged**